arbitrum-docs/for-devs/quickstart-solidity-hardhat.md

```diff
@@ -0,0 +1,302 @@
+
+When our `VendingMachine` contract is deployed to Ethereum, it'll be hosted by Ethereum's decentralized network of nodes. We won't be able to modify the contract's code after it's deployed[^3].
+
+Let's convert our dumb contract into a smart contract.
+
+
+### Configure your project directory
+
+Create a `decentralized-cupcakes` directory for your project. Navigate to this directory from VS Code's integrated terminal and install [hardhat](https://hardhat.org/hardhat-runner/docs/getting-started#overview):
+
+<!-- todo: toggles for npm / yarn? -->
+<!-- todo: save-dev? -->
+<!-- todo: note about windows / wsl / yarn --->
+```
+yarn add hardhat @nomicfoundation/hardhat-toolbox
+```
+
+This installs two packages: `hardhat` lets us test and deploy our smart contracts, and `hardhat-toolbox` is a bundle of popular Hardhat plugins that we'll use later.
+
+Next, run `npx hardhat` to configure Hardhat. Select `Create a JavaScript project` when prompted. Make sure you specify your `decentralized-cupcakes` directory as the project root when asked.
+
+At this point, you should see the following items (among others) in your `decentralized-cupcakes` project directory:
+
+| Item                | Description                                                                                               |
+| ------------------- | --------------------------------------------------------------------------------------------------------- |
+| `contracts/`        | Contains your smart contracts. You should see the `Lock.sol` contract here.                               |
+| `scripts/`          | Contains scripts that you can use to interact with your smart contracts. You should see `deploy.js` here. |
+| `test/`             | Contains test files for your smart contracts. We can use `Lock.js` to test `Lock.sol`.                    |
+| `hardhat.config.js` | Contains the configuration settings for Hardhat.                                                          |
+
+
+Run `npx hardhat compile` to compile the default `contracts`. You may be prompted to install additional dependencies - follow those instructions until this command runs successfully[^4]. You should see `Compiled 1 Solidity file successfully` in the terminal output. You should also see a new `decentralized-cupcakes/artifacts/` directory. This directory contains the compiled smart contract[^5].
+
+Delete the `test/` directory; testing is beyond the scope of this quickstart. Open `scripts/deploy.js` and replace its contents with the following:
+
+```javascript title="scripts/deploy.js"
+const hre = require("hardhat");
+
+async function main() {
+
+  const VendingMachineFactory = await hre.ethers.getContractFactory("VendingMachine");
+  const vendingMachine = await VendingMachineFactory.deploy();
+  await vendingMachine.deployed();
+
+  console.log(
+    `Cupcake vending machine deployed to ${vendingMachine.address}`
+  );
+}
+
+main().catch((error) => {
+  console.error(error);
+  process.exitCode = 1;
```

Comment:
I believe this needs the additional line `process.exit()`

Or alternatively, just replace both w/ the the one-line

`process.exit(1)`